pkgs/applications/virtualization/distrobox/default.nix

```diff
@@ -21,7 +21,7 @@ stdenvNoCC.mkDerivation (finalAttrs: {
 
     # https://github.com/89luca89/distrobox/issues/408
     substituteInPlace ./distrobox-generate-entry \
-      --replace-fail 'icon_default="''${HOME}/.local' "icon_default=\"$out"
+      --replace-fail 'icon_default="''${XDG_DATA_HOME:-''${HOME}/.local' 'icon_default="''${XDG_DATA_HOME:-''$out'
```

Comment:
The literal result of this is:

```bash
icon_default="${XDG_DATA_HOME:-$out/share}/icons/terminal-distrobox-icon.svg"
```

In general, I don't really think we want to inject a Nix store path here. Why don't we just patch this to use the icon name (equivalent to having `Icon=terminal-distrobox-icon` in the final desktop entry)

Reply:
Nevermind, I see that the remaining code assumes that this variable is a path.

I think this whole file should just be patched to resolve the fallback icon using its name.

Reply:
I think this should work:

```patch
diff --git a/pkgs/applications/virtualization/distrobox/default.nix b/pkgs/applications/virtualization/distrobox/default.nix
index f79d09dc94b3..ce3c4e075a95 100644
--- a/pkgs/applications/virtualization/distrobox/default.nix
+++ b/pkgs/applications/virtualization/distrobox/default.nix
@@ -16,12 +16,14 @@ stdenvNoCC.mkDerivation (finalAttrs: {
 
   nativeBuildInputs = [ makeWrapper ];
 
+  patches = [
+    # https://github.com/89luca89/distrobox/issues/408
+    ./relative-default-icon.patch
+  ];
+
   installPhase = ''
     runHook preInstall
 
-    # https://github.com/89luca89/distrobox/issues/408
-    substituteInPlace ./distrobox-generate-entry \
-      --replace-fail 'icon_default="''${XDG_DATA_HOME:-''${HOME}/.local' 'icon_default="''${XDG_DATA_HOME:-''$out'
     ./install -P $out
 
     runHook postInstall
diff --git a/pkgs/applications/virtualization/distrobox/relative-default-icon.patch b/pkgs/applications/virtualization/distrobox/relative-default-icon.patch
new file mode 100644
index 000000000000..069c2d40d618
--- /dev/null
+++ b/pkgs/applications/virtualization/distrobox/relative-default-icon.patch
@@ -0,0 +1,26 @@
+diff --git a/distrobox-generate-entry b/distrobox-generate-entry
+index 65fcea0..8d57e4e 100755
+--- a/distrobox-generate-entry
++++ b/distrobox-generate-entry
+@@ -51,7 +51,7 @@ container_manager="autodetect"
+ container_name_default="my-distrobox"
+ delete=0
+ icon="auto"
+-icon_default="${XDG_DATA_HOME:-${HOME}/.local/share}/icons/terminal-distrobox-icon.svg"
++icon_default="terminal-distrobox-icon"
+ verbose=0
+ online=0
+ version="1.8.0"
+@@ -335,12 +335,6 @@ if [ "${icon}" = "auto" ]; then
+ 
+ 	icon_url="$(echo "${DISTRO_ICON_MAP}" | grep "${container_distro}:" | cut -d':' -f2-)"
+ 
+-	# Distro not found in our map, fallback to generic icon
+-	if [ -z "${icon_url}" ]; then
+-		icon_url="https://raw.githubusercontent.com/89luca89/distrobox/main/icons/terminal-distrobox-icon.svg"
+-		container_distro="terminal-distrobox-icon"
+-	fi
+-
+ 	if [ -n "${icon_url}" ] && [ "${download}" != "null" ]; then
+ 		icon_extension="${icon_url##*.}"
+ 
```